ext/drreg/drreg.dox

```diff
@@ -42,7 +42,11 @@ instrumentation pass scheme.  The results are most efficient when used in
 the \p drmgr instrumentation insertion phase.  Use outside of the insertion
 phase, such as during \p drmgr's instrumentation-to-instrumentation
 transformation phase, or use outside of \p drmgr entirely, is supported,
-but duplicate spills and restores may not be optimized away.  When using
+but duplicate spills and restores may not be optimized away.  Also, for
+non-insertion phases, the responsibility to restore a spilled app value
+before an app read, and to re-spill the app value after an app write, lies
```

Comment:
So we would also want a manual restore before a label with note DR_NOTE_ANNOTATION which my PR should have added to the docs: if you could add that here.

Reply:
Yes, a manual restore is required in app2app before that label.

Note that the insertion phase will re-spill the above restored value to its slot only if it sees that the value is live. This is of course always true when there's an app read coming up. For user prompted restores (like drreg_get_app_value or drreg_statelessly_restore_app_value), app2app needs to add a dummy read -- see PR #5126 for an example. Is anything like that required for the annotation case too?

Reply:
I believe that having every app2app drreg usage that crosses app instructions look for the annotation label and restore prior to that label is sufficient.

Reply:
Done. PTAL.